protocol/message.go

```diff
@@ -0,0 +1,74 @@
+package protocol
+
+import (
+	m "github.com/coniks-sys/coniks-go/merkletree"
+)
+
+// Defines constants representing the types
+// of messages exchanged by clients and servers.
+const (
+	RegistrationType = iota
+	KeyLookupType
+	ConsistencyCheckType
+)
+
+type Request struct {
+	Type    int
+	Request interface{}
+}
+
+type Response interface{}
+
+type ErrorResponse struct {
+	Error ErrorCode
+}
+
+func NewErrorResponse(e ErrorCode) Response {
+	return &ErrorResponse{Error: e}
+}
+
+type RegistrationRequest struct {
+	Username               string `json:"username"`
+	Key                    []byte `json:"key"`
+	AllowUnsignedKeychange bool   `json:"allow_unsigned_key_change,omitempty"`
+	AllowPublicLookup      bool   `json:"allow_public_lookup,omitempty"`
+}
+
+type RegistrationProof struct {
+	Type int
+	AP   *m.AuthenticationPath
+	STR  *m.SignedTreeRoot
```

Comment:
This makes me think about some changes in 7e807e5821ed0b73622108b2b1b746fc3a42f8ca. What do you think?